actionpack/lib/action_controller/metal/mime_responds.rb

```diff
@@ -288,7 +288,7 @@ def method_missing(name, *args, &block)
         end
 
         def variant
-          if @variant.nil?
+          if @variant.empty?
             @variants[:none] || @variants[:any]
           elsif (@variants.keys & @variant).any?
             @variant.each do |v|
```

Comment:
Since we're already touching this code, maybe we can improve it. How about this? :smile:

```ruby
def variant
  if @variant.empty?
    @variants[:none] || @variants[:any]
  else
    @variants[ @variant.find { |v| @variants.key?(v) } || :any ]
  end
end
```